compiler/rustc_hir_analysis/src/check/fn_ctxt/suggestions.rs

```diff
@@ -80,6 +80,38 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let Some((def_id_or_name, output, inputs)) = self.extract_callable_info(expr, found)
             else { return false; };
         if can_satisfy(output) {
+            // suggest `unwrap_or_else` for `Option` and `Result`, when a closure is given
+            if let hir::ExprKind::Closure(..) = expr.kind &&
```

Comment:
If you use `extract_callable_info` then you don't need to also check if the expr is a closure.